lib/ansible/modules/cloud/amazon/elasticsearch.py

```diff
@@ -0,0 +1,377 @@
+    region: us-west-1
+"""
+
+RETURN = """
+arn:
+    description: ARN of Elasticsearch cluster.
+    returned: always
+    type: string
+    sample: "arn:aws:es:us-west-1:000000000000:domain/testing"
+endpoint:
+    description: When cluster is finished being created, endpoint to submit requests to.
+    returned: always
+    type: string
+    sample: "search-testing-asdfqwerty.us-west-1.es.amazonaws.com"
+processing:
+    description: Is the cluster processing config changes.
+    returned: always
+    type: bool
+elasticsearch_version:
+    description: Which version of Elasticsearch cluster is running.
+    returned: always
+    type: string
+    sample: "5.1"
+instance_type:
+    description: Instance type of non-master nodes in the cluster.
+    returned: always
+    type: string
+    sample: "t2.medium.elasticsearch"
+instance_count:
+    description: Number of non-master nodes in the cluster.
+    returned: always
+    type: int
+dedicated_master_enabled:
+    description: Does cluster have dedicated master nodes.
+    returned: always
+    type: bool
+zone_awareness:
+    description: Is cluster aware of AWS availability zones.
+    returned: always
+    type: bool
+dedicated_master_type:
+    description: Instance type of dedicated master nodes.
+    returned: always
+    type: string
+    sample: "t2.medium.elasticsearch"
+dedicated_master_count:
+    description: Number of dedicated master nodes.
+    returned: always
+    type: int
+ebs_volume_type:
+    description: What EBS volume type nodes have.
+    returned: always
+    type: string
+    sample: "gp2"
+ebs_volume_size:
+    description: What size EBS volume each node has.
+    returned: always
+    type: int
+access_policies:
+    description: JSON string of the access policy of the cluster.
+    returned: always
+    type: string
+    sample: '{"Statement": [{"Action": "*", "Resource": "arn:aws:iam::0000000000000:user/you", "Effect": "Allow"}]}'
+"""
+
+import collections
+import json
+import time
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils import ec2
+
+try:
+    import botocore.exceptions
+except ImportError:
+    pass  # will be detected by ec2.HAS_BOTO3
+
+
+def _create_elasticsearch_domain(connection, module):
+    elasticsearch_cluster_config = {
+        'InstanceCount': module.params['instance_count'],
+        'InstanceType': module.params['instance_type'],
+        'ZoneAwarenessEnabled': module.params['zone_awareness'],
+        'DedicatedMasterEnabled': module.params['dedicated_master_enabled'],
+    }
+
+    if module.params['dedicated_master_enabled']:
+        elasticsearch_cluster_config['DedicatedMasterType'] = module.params['dedicated_master_type']
+        elasticsearch_cluster_config['DedicatedMasterCount'] = module.params['dedicated_master_count']
+
+    connection.create_elasticsearch_domain(
```

Comment:
This needs to handle ClientError and BotoCoreError. See the guidelines if you haven't already: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#exception-handling-for-boto3-and-botocore.

ClientError has a .response and BotoCoreError does not. So it could look something like:
```
try:
    connection.create_elasticsearch_domain(...)
except ClientError as e:
    module.fail_json(msg="Unable to create elasticsearch domain: {0}".format(to_native(e)),
                     exception=traceback.format_exc(),
                     **camel_dict_to_snake_dict(e.response))
except BotoCoreError as e:
    module.fail_json(msg="Unable to create elasticsearch domain: {0}".format(to_native(e)),
                     exception=traceback.format_exc())
```
That will require `import traceback` and `from ansible.module_utils.ec2 import camel_dict_to_snake_dict` and `from ansible.module_utils._text import to_native`


Or: 

you can import AnsibleAWSModule from ansible.module_utils.aws.core to replace AnsibleModule, and then you can do:

```
try:
    connection.create_elasticsearch_domain(...)
except (ClientError, BotoCoreError) as e:
    module.fail_json_aws(e, msg="Unable to create elasticsearch domain: {0}".format(to_native(e)))
```
(also needing `from ansible.module_utils._text import to_native`)

This would also let you remove the check for HAS_BOTO3 since AnsibleAWSModule handles that. :)